File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.js
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     constructor(val = 0, next = null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */

class Solution {
    /**
     * @param {ListNode} head
     * @return {ListNode}
     */
    reverseList(head) {
        let prev = null;
        let iter = head;

        while(iter){
            let temp = iter.next;

            iter.next = prev;
            prev = iter;

            iter = temp;
        }

        return prev;
    }
}

File: Data Structures & Algorithms/reorder-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reorderList(self, head: Optional[ListNode]) -> None:
        slow = head
        fast = head.next

        while fast and fast.next:
            fast = fast.next.next
            slow = slow.next

        fast = slow.next
        slow.next = None

        prev = None
        curr = fast

        while curr:
            temp = curr.next
            curr.next = prev
            prev = curr
            curr = temp

        l1 = head
        l2 = prev

        while l1 and l2:
            temp = l1.next
            l1.next = l2

            temp2 = l2.next
            l2.next = temp

            l1 = temp;
            l2 = temp2;







